### ISSUE TITLE
Automatic solver selection for DynamicalODEProblem gives incorrect results

### ISSUE THREAD

**Issue by samuela** [SciML/DifferentialEquations.jl#681]
I realized recently that the automatic solver selection thing for `DynamicalODEProblem`s gives incorrect integration results. In the following example we accumulate a loss into `x[1]` which should integrate to 2. However, with the default solver selection the result is `0.0`:

```julia
import DifferentialEquations: Tsit5, DynamicalODEProblem, solve

v0 = zeros(2)
x0 = ones(2)

aug_dyn_v(v_aug, x_aug, p, t) = zeros(3)
aug_dyn_x(v_aug, x_aug, p, t) = begin
    v = v_aug[2:end]
    x = x_aug[2:end]
    # We accumulate a "cost" metric in x[1], and use x[2:end] for the actual state.
    [sum(x .^ 2); v]
end

# automatic solver selection => loss = 0.0
solvealg = nothing

# Tsit5 => loss > 0.0 (as expected)
# solvealg = Tsit5()

sol = solve(
    DynamicalODEProblem(
        aug_dyn_v,
        aug_dyn_x,
        [0.0; v0],
        [0.0; x0],
        (0.0, 1.0),
    ),
    solvealg,
)

loss = sol.u[end].x[2][1]
@assert loss > 0
```

**Comment by ChrisRackauckas** [comment]
Yeah, that's not good. IIRC Runge-Kutta-Nystrom requires a second order ODE, at least how it's implemented, so we should check the second order ODE trait before defaulting to it. I thought that was theoretically relaxed though, but it might've been a shortcut in the implementation which we should probably fix up.

@YingboMa double chime in here.

**Comment by samuela** [comment]
I just tested this on `[0c46a032] DifferentialEquations v6.19.0` and the issue appears to be gone. Has this been resolved?

**Comment by ChrisRackauckas** [comment]
Yes
